### PR TITLE
fix: use xargs to speed up chown during init

### PR DESCRIPTION
**Pull request by Daniel-Aaron-Bloom**
`init-code-server` currently uses `find -exec` which accumulates all files before `chown`ing them.

This is slow for users with a lot of files. Using `xargs` makes it much faster.

[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-code-server/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
- `find -print0` and `xargs -0` change everything to use null characters instead of whitespace between paths
- `-n 1000` is the number of files to `chown` per xarg execution
  - Performance capped once I got above ~20 and stayed there even up to 1 million, 1000 seemed a nice middle-ground between those which should maximize performance across different machine specs.
- `-P $(nproc)` set the parallelism to the number of cores

### Old
```bash
$ time find /config -path /config/workspace -prune -o -exec chown abc:abc {} +

real    12m19.069s
user    0m1.601s
sys     2m33.088s
```

### New
```bash
$ time find /config -path /config/workspace -prune -o -print0 | xargs -0 -n 1000 -P $(nproc) chown abc:abc

real    0m46.416s
user    0m2.339s
sys     2m13.735s
```

## How Has This Been Tested?
Manually by modifying the init script when building my image.

### CODE REVIEW

**Comment by thespad** [comment]
Oh sorry, still half asleep, didn't realise we had an errant hand-crafted chown still in the code-server init. Most of our chowns are handled by https://github.com/linuxserver/docker-mods/blob/mod-scripts/lsiown.v1, which does the same thing as the line you've modified.

**Comment by thespad** [comment]
TL;DR we can't remember, but I think it was an issue with low-powered devices performing very badly with the xargs parallel chowns, especially those with slow storage.

One question I have is what have you got in your /config mount (outside of /config/workspace) that's taking *12 minutes* to chown?

**Comment by aptalca** [comment]
@Daniel-Aaron-Bloom can you give this test image a try?
`ghcr.io/linuxserver/lspipepr-code-server:4.93.1-pkg-3d426902-dev-42aae8bde4a11d239abb91e5fd75f0e843ecc8a9-pr-188`
It's the result of #188 , which should ignore the dev/cache folders during chown

**Comment by aptalca** [comment]
scratch that, we decided to make chown conditional instead
